tests/visualization_tests/test_pareto_front.py

```diff
@@ -31,35 +34,78 @@ def test_plot_pareto_front_2d(
+            if axis_order is None:
+                assert (figure.data[2]["x"] + figure.data[1]["x"] + figure.data[0]["x"]) == data[0]
+                assert (figure.data[2]["y"] + figure.data[1]["y"] + figure.data[0]["y"]) == data[1]
+            else:
+                assert (figure.data[2]["x"] + figure.data[1]["x"] + figure.data[0]["x"]) == data[
+                    axis_order[0]
+                ]
+                assert (figure.data[2]["y"] + figure.data[1]["y"] + figure.data[0]["y"]) == data[
+                    axis_order[1]
+                ]
         else:
-            assert (figure.data[1]["x"] + figure.data[0]["x"]) == data[axis_order[0]]
-            assert (figure.data[1]["y"] + figure.data[0]["y"]) == data[axis_order[1]]
+            # The enqueue order of trial is: infeasible, feasible.
+            data = [(1, 0, 1), (1, 2, 0)]  # type: ignore
+            if axis_order is None:
+                assert (figure.data[2]["x"] + figure.data[1]["x"] + figure.data[0]["x"]) == data[0]
+                assert (figure.data[2]["y"] + figure.data[1]["y"] + figure.data[0]["y"]) == data[1]
+            else:
+                assert (figure.data[2]["x"] + figure.data[1]["x"] + figure.data[0]["x"]) == data[
+                    axis_order[0]
+                ]
+                assert (figure.data[2]["y"] + figure.data[1]["y"] + figure.data[0]["y"]) == data[
+                    axis_order[1]
+                ]
     else:
-        data = [(1, 0), (0, 1)]  # type: ignore
-        if axis_order is None:
-            assert (figure.data[1]["x"] + figure.data[0]["x"]) == data[0]
-            assert (figure.data[1]["y"] + figure.data[0]["y"]) == data[1]
+        assert len(figure.data) == 2
+        if include_dominated_trials:
+            # The last elements come from dominated trial that is enqueued firstly.
+            data = [(0, 1, 1, 1), (2, 0, 2, 1)]  # type: ignore
+            if axis_order is None:
+                assert (figure.data[1]["x"] + figure.data[0]["x"]) == data[0]
+                assert (figure.data[1]["y"] + figure.data[0]["y"]) == data[1]
+            else:
+                assert (figure.data[1]["x"] + figure.data[0]["x"]) == data[axis_order[0]]
+                assert (figure.data[1]["y"] + figure.data[0]["y"]) == data[axis_order[1]]
         else:
-            assert (figure.data[1]["x"] + figure.data[0]["x"]) == data[axis_order[0]]
-            assert (figure.data[1]["y"] + figure.data[0]["y"]) == data[axis_order[1]]
+            data = [(0, 1), (2, 0)]  # type: ignore
+            if axis_order is None:
+                assert (figure.data[1]["x"] + figure.data[0]["x"]) == data[0]
+                assert (figure.data[1]["y"] + figure.data[0]["y"]) == data[1]
+            else:
+                assert (figure.data[1]["x"] + figure.data[0]["x"]) == data[axis_order[0]]
+                assert (figure.data[1]["y"] + figure.data[0]["y"]) == data[axis_order[1]]
```

Comment:
These lines for data checking seem to be deeply nested. Do you think if it is possible to extract it as a new test function to reduce the nest level?